src/listener/can_open_listener.py

```diff
@@ -87,10 +89,17 @@ async def async_network_loop(self):
             Loops forever.
         '''
         try:
-            while True:
-                self.listen_to_network()
-                # Sleeps 0.1 seconds
-                await asyncio.sleep(0.1)
+            self.network.scanner.search()
+            time.sleep(0.05)
+            if(len(self.network.scanner.nodes) > 0):
+                while True:
+                    self.listen_to_network(self.network.scanner.nodes)
+                    # Sleeps 0.1 seconds
+                    await asyncio.sleep(0.1)
+            else:
+                logging.error('No nodes to listen to!')
+        except can.CanError:
```

Comment:
Ik had in mijn code nog een if statement, just in case CanError iets anders kon raisen dan "Network is down"

```
        try:
            #run a search to test network connection
            self.network.scanner.search()
        except CanError as exc:
            if "[Errno 100] Network is down" in repr(exc):
                print("CAN network down. Attempting to initialise CAN network\n")

                os.system("sudo ip lin set can0 up type can bitrate 500000")
                os.system("sudo ifconfig can0 txqueuelen 10000")
                print("CAN network up")
            else:
                raise exc
```

Reply:
aangepast